projects/core/src/cart/store/effects/cart-entry.effect.ts

```diff
@@ -82,6 +89,57 @@ export class CartEntryEffects {
     )
   );
 
+  @Effect()
+  removeEntryByKey$: Observable<
+    | CartActions.CartRemoveEntryByKeySuccess
+    | CartActions.CartRemoveEntryByKeyFail
+    | DeprecatedCartActions.LoadCart
+  > = this.actions$.pipe(
+    ofType(CartActions.CART_REMOVE_ENTRY_BY_KEY),
+    map((action: CartActions.CartRemoveEntryByKey) => action.payload),
+    concatMap(payload =>
+      // TODO add take until for contextChange
+      this.cartConnector.load(payload.userId, payload.cartId).pipe(
+        switchMap(cart => {
+          const entryToRemove = cart.entries.find(
+            entry => entry.key === payload.key
+          );
+          return this.cartEntryConnector
+            .remove(
+              payload.userId,
+              payload.cartId,
+              `${entryToRemove.entryNumber}`
+            )
+            .pipe(
+              mergeMap(() => {
+                return [
+                  new CartActions.CartRemoveEntryByKeySuccess({
+                    ...payload,
```

Comment:
With the introduction of the event system we identified often a poor payload for cart related actions. Ideally we have a cart summary for cart operations. Is this something we can do in this ticket, or in a follow up?

Reply:
I wouldn't mix those two issues. I try to provide better payload in new actions but touching the old ones is kinda ugly. Especially with fail actions that have this error at the top level. We would need to provide extra payload as a second parameter and that would be another inconsistency in the codebase. 

I added a new point to #5234 to look into cart payloads.

I would just start with the recommendation for new actions to always use objects for payload (more consistent and not breaking) and to provide in success/fail actions payload everything that was passed to the initial action.